pkg/ccl/changefeedccl/kvevent/bench_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package kvevent_test
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/kvevent"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql/rowenc/keyside"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkMemBuffer(b *testing.B) {
+	rand, _ := randutil.NewTestRand()
+
+	run := func() {
+		numRows := 2 << 20
+		ba, release := getBoundAccountWithBudget(4096)
+		defer release()
+
+		metrics := kvevent.MakeMetrics(time.Minute)
+
+		st := cluster.MakeTestingClusterSettings()
+		buf := kvevent.NewMemBuffer(ba, &st.SV, &metrics)
+		defer func() {
+			require.NoError(b, buf.CloseWithReason(context.Background(), nil))
+		}()
+
+		producerCtx, stopProducers := context.WithCancel(context.Background())
+		wg := ctxgroup.WithContext(producerCtx)
+		defer func() {
+			_ = wg.Wait() // Ignore error -- this group returns context cancellation.
+		}()
+
+		wg.GoCtx(func(ctx context.Context) error {
```

Comment:
On the one hand, you're not testing the "blocking" portion of the blocking buffer; so, you could just do {Add; followed by Get}.
If you want to see the savings just due to your change (i.e ResolvedSpan -> &ResolvedSpan), then you could just do that...

On the other hand, if you don't put anything in, then you're not actually exercising e.g linked list alloc.  I would be interesting to see just how full you can get this buffer. 

You could also consider simplying this benchmark somewhat.  You could see how much memory each event uses -- and just put in fixed number of events.
Maybe 2M like you have, maybe less... And just put it all in; and then drain it all out...

Reply:
Actually, this allocations due to linked list -- perhaps better be tested as a follow on when you switch to chunk allocation.

Reply:
I updated the benchmark code to fill the buffer and empty it, while putting as many events as possible until it is full. I like this because it captures event allocs and linked list allocs. It also simulates something closer to what we see in production because the buffer actually fills up. I updated the results in the PR message and they are still good.